src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((prev, curr) => {
+    const itemKey = curr[key] as string;
+
+    return {
+      ...prev,
+      [itemKey]: prev[itemKey]
+        ? [...prev[itemKey], curr]
+        : [curr],
+    };
```

Comment:
let's make it simpler 
```suggestion
    prev[itemKey] 
       ? prev[itemKey].push(curr)
       : prev[itemKey] = [curr] 
    return prev;
```

Reply:
This way does not work:(
![image](https://github.com/mate-academy/ts_group_by_key/assets/96629199/249d2788-0b1d-4681-8b6a-dc723cb9c60f)

Reply:
but in VS Code on js it works!

Reply:
Okay, let's leave the previous version